$NetBSD: patch-9045,v 1.1 2014/11/18 14:38:15 manu Exp $
From fb36dbea2f492317a744b828d8ba8bdaab78ae3a Mon Sep 17 00:00:00 2001
From: Emmanuel Dreyfus <manu@netbsd.org>
Date: Tue, 4 Nov 2014 15:57:59 +0100
Subject: [PATCH] Portability fix: mount.glusterfs

Remove bash-specific syntax from mount.glusterfs

This is a backport of  Iec3a52686f7cee1825ac5a06c11fb8ac4d3e5d65

BUG: 1138897
Change-Id: I3149cd4d94c862a3857709c00cf0ed0377976f40
Signed-off-by: Emmanuel Dreyfus <manu@netbsd.org>
---
 xlators/mount/fuse/utils/mount_glusterfs.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git xlators/mount/fuse/utils/mount_glusterfs.in xlators/mount/fuse/utils/mount_glusterfs.in
index e0a06ea..dc326b4 100755
--- xlators/mount/fuse/utils/mount_glusterfs.in
+++ xlators/mount/fuse/utils/mount_glusterfs.in
@@ -449,7 +449,7 @@ without_options()
 parse_options()
 {
     local optarg=${1}
-    for pair in $(echo ${optarg//,/ }); do
+    for pair in $(echo $optarg | sed 's/,/ /g'); do
         key=$(echo "$pair" | cut -f1 -d'=');
         value=$(echo "$pair" | cut -f2- -d'=');
         if [ "$key" = "$value" ]; then
@@ -463,7 +463,8 @@ parse_options()
 main ()
 {
     ## `mount` on OSX specifies options as first argument
-    if [[ $1 =~ "-o" ]]; then
+    echo $1|grep -q -- "-o"
+    if [ $? -eq 0 ];  then
         volfile_loc=$3
         mount_point=$4
     else
-- 
1.8.2.3

